R Guide/R Kaggle Guide (Titanic).Rmd

```diff
@@ -0,0 +1,96 @@
+---
+title: "R Kaggle Guide (Titanic)"
+author: "UWaterloo Data Science Club"
+date: "August 18, 2017"
+output: pdf_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+This guide is based on R 3.3.2. We recommend downloading R [here](https://r-project.org/) along with [R Studio](https://www.rstudio.com/products/rstudio/download/), a set of integrated tools that will make your life a lot easier. This guide assumes that you have some sort of programming experience.
+
+This guide is written in something called R markdown, which allows us to describe our process while showing and executing code (kind of like a notebook). This notebook process is type of what Kaggle calls a **kernel**. When working in R Studio, pressing `ctrl + enter` will run the current line of code.
+
+---
+This guide will look at the Titanic dataset, we will see if we can predict what types of people would have survived on the Titanic.
+
+So first we will import some useful libraries. R is old and there are confusing things about the language that came up over time, the tidyverse stack is a set of libraries that make these functions more consistent and powerful.
+```{R}
+library("tidyverse")
+```
+
+Note the conflict errors indicate that two different libraries have the same function name. We don't need to worry about this for now. Now we can import our data into a dataframe.
+
+```{R, echo=FALSE}
+titanic_data <- read_csv("../titanic_data/train.csv")  # .. indicates the parent folder
+```
+
+The code output tells us about how each column was imported such as what data type is stored. To understand more about what options you have, you can type `?read_csv` in the console, or `?<function_name>` for any function. If you don't know what exactly the function name is, you can do `??<query>` which will return you all manual pages relevant to the query. 
+
+Normally we won't worry to much about datatypes, but notice how certain columns like `Survived` and `Pclass` were imported as integers? The problem is that we use integers to differentiate the data value but there isn't any ineherent order to the numbers. Instead we can convert integers, characters, etc.to categories, which is called a **factor** in R.
+
+The `$` let's us select specific variables in a dataframe.
+
+```{R}
+titanic_data$Survived <- as.factor(titanic_data$Survived)  
+titanic_data$Pclass <- as.factor(titanic_data$Pclass)
+titanic_data$Sex <- as.factor(titanic_data$Sex)
+titanic_data$Embarked <- as.factor(titanic_data$Embarked)
+```
+
+We can observe the first `n` entries of our dataframe by using the `head()` function, likewise we to observe the last `n` entires we can use `tail()`. If there are too many variables, the output will omit them to save space.
+
+```{R}
+head(titanic_data, 5)
```

Comment:
I'd love for you to introduce the `%>%` operator just because its preferred way of doing things.

perhaps explain what it does, and show that you can do both 

`head(df, 5)` and `df %>% head(5)`